src/PositionManager.sol

```diff
@@ -154,9 +154,9 @@ contract PositionManager is IPositionManager, ERC721Permit, PoolInitializer, Mul
 
         // the sender is the payer or receiver
         if (currencyDelta < 0) {
-            currency.settle(poolManager, sender, uint256(-int256(currencyDelta)), false);
+            poolManager.settle(currency, sender, uint256(-int256(currencyDelta)));
```

Comment:
I kind of preferred doing currency.settle bc its very clear its not actually calling the manager contract but rather its using a library wrapping the manager.settle call

Reply:
i think this is still not fixed?

Reply:
i see that this got fixed.. still keft some comments tho abt some changes re DeltaResolver lib below!